core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -73,6 +73,20 @@ private Optional<Result> createGetResult(Snapshot.Key key, List<String> projecti
   }
 
   public List<Result> scan(Scan scan) throws CrudException {
+    List<Result> results = scanInternal(scan);
+
+    if (ScalarDbUtils.isRelational(scan)) {
+      // We verify if this scan does not overlap previous writes using the results obtained from
+      // the actual scan since the condition (i.e., where clause) is arbitrary in the relational
+      // scan, and thus, the write command may not have columns used in the condition, which are
+      // necessary to determine overlaps.
+      snapshot.verify(scan);
+    }
```

Comment:
The other major change is starting from here. This prevents scan-after-write with a bit different approach (i.e., use actual scan results) in existing scans since we need to handle the case that the user's `Put` does not have columns required to check if they match the `Scan` conditions.